libs/wingsdk/src/target-sim/bucket.inflight.ts

```diff
@@ -227,43 +234,52 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
    * @param key Key of the object.
    * @throws if the object does not exist.
    */
-  public async metadata(key: string): Promise<ObjectMetadata> {
+  public async getMetadata(key: string): Promise<ObjectMetadata> {
```

Comment:
@Chriscbr I don't think we've established these API naming guidelines yet, but my preference is not to use `getXxx()` for getters, just `xxxx()`. What do you think?

```suggestion
  public async metadata(key: string): Promise<ObjectMetadata> {
```

Reply:
For me it's actually confusing to not have verbs to define methods since that's what I've always seen as [common best practice](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods) in OOP. Besides, you have a `Bucket.get()` method to get the object, but then you don't want a `Bucket.getMetadata()` to get the object's metadata? At first I thought that was an attribute, not a method. I had the exact same issue when using the `cloud.Secret`, where I thought `value()` was an attribute and not a method.

Reply:
If you don't want to use getters altogether, then `Bucket.get()` should be renamed to `Bucket.object()` for consistency, which ends up becoming as confusing as `Secret.value()`.

Reply:
I tend to agree with @eladb here - I think we can leave the method as `metadata()` for now. Longer, precise names like "getObject()" or "getMetadata()" can sometimes bring more consistency to our APIs, but we have to balance it with the benefits of making our APIs convenient to read and write. In this case, I don't feel like changing `metadata()` to `getMetadata()` clarifies what the method does in a significant way. But it's definitely something we can revisit if a lot of users get stuck on this...